packages/preferences/src/browser/views/components/single-preference-wrapper.tsx

```diff
@@ -0,0 +1,181 @@
+ ********************************************************************************/
+
+import * as React from 'react';
+import { Menu, PreferenceDataProperty, PreferenceScope, PreferenceItem, PreferenceService } from '@theia/core/lib/browser';
+import { PreferenceSelectInput, PreferenceBooleanInput, PreferenceStringInput, PreferenceNumberInput, PreferenceJSONInput, PreferenceArrayInput } from '.';
+import { Preference } from '../../util/preference-types';
+
+interface SinglePreferenceWrapperProps {
+    preferenceDisplayNode: Preference.NodeWithValueInAllScopes;
+    getMenuForPreference(prefID: string, value: PreferenceItem): Menu;
+    currentScope: number;
+    currentScopeURI: string;
+    preferencesService: PreferenceService;
+    openJSON(): void;
+}
+
+interface SinglePreferenceWrapperState {
+    showCog: boolean;
+    menuOpen: boolean;
+}
+
+export class SinglePreferenceWrapper extends React.Component<SinglePreferenceWrapperProps, SinglePreferenceWrapperState> {
+    protected contextMenu: Menu;
+
+    state: SinglePreferenceWrapperState = {
+        showCog: false,
+        menuOpen: false
+    };
+
+    protected handleOnCogClick(e: React.MouseEvent | React.KeyboardEvent, prefID: string, currentValue?: PreferenceItem): void {
+        if (currentValue !== undefined) {
+            const target = (e.target as HTMLElement);
+            const domRect = target.getBoundingClientRect();
+            this.contextMenu = this.props.getMenuForPreference(prefID, currentValue);
+            this.contextMenu.open(domRect.left, domRect.bottom);
+            this.setState({ menuOpen: true });
+            this.contextMenu.aboutToClose.connect(() => {
+                this.setState({ menuOpen: false });
+            });
+        }
+    }
+
+    render(): React.ReactNode {
+        const { preferenceDisplayNode } = this.props;
+        const { preference: { data, values } } = preferenceDisplayNode;
+
+        const value = this.getValueInCurrentScope(values, this.props.currentScope);
+
+        const currentValueIsDefaultValue = value === data.defaultValue;
+
+        const singlePreferenceValueDisplayNode = { ...preferenceDisplayNode, preference: { data, value } };
+        const description = data.markdownDescription || data.description;
+        if (preferenceDisplayNode.visible) {
+            return (<li
+                className='single-pref'
+                id={`${preferenceDisplayNode.id}-editor`}
+                key={preferenceDisplayNode.id}
+                data-id={preferenceDisplayNode.id}
+            >
+                <div className="pref-name">
+                    {preferenceDisplayNode.name}
+                    {this.renderOtherModifiedScopes(values, data, this.props.currentScope)}
+                </div>
+                <div className={`pref-context-gutter ${!currentValueIsDefaultValue ? 'theia-mod-item-modified' : ''}`}
+                    onMouseOver={() => this.setState({ showCog: true })}
```

Comment:
please review all react components that you never use arrow functions, that's perf killer